packages/fdc3-get-agent/test/support/TestServerContext.ts

```diff
@@ -0,0 +1,156 @@
+import { ServerContext, InstanceID } from '@kite9/fdc3-web-impl'
+import { CustomWorld } from '../world'
+import { Context } from '@kite9/fdc3-context'
+import { OpenError, AppIdentifier, AppIntent } from '@kite9/fdc3-standard'
+import { AppRegistration, State } from '@kite9/fdc3-web-impl'
+
+type ConnectionDetails = AppRegistration & {
+    msg?: object
+    connectionId: string,
+    externalPort: MessagePort,
+    internalPort: MessagePort,
+    url: string
+}
+
+type MessageRecord = {
+    to?: AppIdentifier,
+    uuid?: InstanceID,
+    msg: object
+}
+
+export class TestServerContext implements ServerContext<ConnectionDetails> {
+
+    public postedMessages: MessageRecord[] = []
+    private readonly cw: CustomWorld
+    private instances: ConnectionDetails[] = []
+    private nextInstanceId: number = 0
+    private nextUUID: number = 0
+
+    constructor(cw: CustomWorld) {
+        this.cw = cw
+    }
+
+    async narrowIntents(_raiser: AppIdentifier, appIntents: AppIntent[], _context: Context): Promise<AppIntent[]> {
+        return appIntents
+    }
+
+    getInstanceDetails(uuid: string) {
+        return this.instances.find(ca => ca.instanceId === uuid)
+    }
+
+    setInstanceDetails(uuid: InstanceID, appId: ConnectionDetails) {
+        this.instances = this.instances.filter(ca => ca.connectionId !== uuid)
+        this.instances.push({
+            ...appId,
+            connectionId: uuid
+        })
+    }
+
+    getMatchingInstance(url: string): ConnectionDetails | undefined {
+        return this.instances.find(ca => ca.url === url)
+    }
+
+    async shutdown(): Promise<void> {
+        await Promise.all(this.instances.map(i => i.internalPort.close()))
+        await Promise.all(this.instances.map(i => i.externalPort.close()))
+    }
+
+    async open(appId: string): Promise<InstanceID> {
+        const ni = this.nextInstanceId++
+        if (appId.includes("missing")) {
+            throw new Error(OpenError.AppNotFound)
+        } else {
+            const mc = new MessageChannel()
+            const internalPort = mc.port1
+            const externalPort = mc.port2;
+
+            (internalPort as any).name = "internalPort-" + ni;
+            (externalPort as any).name = "externalPort-" + ni;
+
+            internalPort.start()
+
+            const connectionDetails = {
+                appId,
+                instanceId: "uuid-" + ni,
+                connected: false,
+                connectionId: "uuid-" + ni,
+                externalPort,
+                internalPort,
+                url: "https://dummyOrigin.test/path",
+                state: State.Pending
+            }
+
+            this.instances.push(connectionDetails)
+            internalPort.onmessage = (msg) => {
+                console.log(`Received message on internalPort ${appId}: ${JSON.stringify(msg.data)}`)
```

Comment:
Did you intend to leave this console.log line? Either add a comment to be explicit, or use console.debug

Reply:
Done in #1424